### PR TITLE
Use pip install instead of git clone to retrieve pyperformance.

### PR DESCRIPTION
**Pull request by dvalinrh**

### CODE REVIEW

File: pyperf/pyperf_run [hunk 1]
```diff
@@ -16,6 +17,7 @@ exit_out()
 usage()
 {
 	echo "$1 Usage:"
+	echo "--pyperf_version <vers number>: Versionof pyperf to run, default is $PYPERF_VERSION."
```
Comment (kdvalin) [inline pyperf/pyperf_run:20]:
```suggestion
	echo "--pyperf_version <vers number>: Version of pyperf to run, default is $PYPERF_VERSION."
```

File: pyperf/pyperf_run [hunk 2]
```diff
@@ -4,6 +4,7 @@ PATH="${PATH}:/usr/local/bin"
 export PATH
 python_pkgs=""
 python_exec=""
+PYPERF_VERSION=1.0.4
```
Comment (kdvalin) [inline pyperf/pyperf_run:7]:
```suggestion
PYPERF_VERSION="1.0.4"
```

File: pyperf/pyperf_run [hunk 3]
```diff
@@ -245,15 +252,11 @@ done
 
 if [ $to_pbench -eq 0 ]; then
 	rm -rf pyperformance
-	git clone https://github.com/python/pyperformance
+	python3 -m pip install pyperformance==$PYPERF_VERSION
```
Comment (kdvalin) [inline pyperf/pyperf_run:255]: We should be making sure that `pip` is installed.  Either through `python3 -m ensurepip --upgrade` or by specifying `python3-pip` as a dependency.

**Review by kdvalin — CHANGES_REQUESTED** [review]

File: pyperf/pyperf_run [hunk 4]
```diff
@@ -245,15 +252,18 @@ done
...
+		exit_out "Failed: python3 -m ensurepip --upgrade" 1
 	fi
-	cd pyperformance
-	git checkout tags/1.0.4
+	python3 -m pip install pyperformance==$PYPERF_VERSION
 	if [ $? -ne 0 ]; then
-		exit_out "Checkout of 1.0.4 failed." 1
+		exit_out "python3 -m pip install pyperformance==$PYPERF_VERSION: failed" 1
 	fi
+	cd pyperformance
```
Comment (kdvalin) [inline pyperf/pyperf_run:266]: Is this still needed?  Aren't we removing the clone that creates the `pyperformance` dir?

File: pyperf/pyperf_run [hunk 5]
```diff
@@ -16,6 +17,7 @@ exit_out()
 usage()
 {
 	echo "$1 Usage:"
+	echo "--pyperf_version <vers number>: Version of pyperf to run, default is $PYPERF_VERSION."
```
Comment (kdvalin) [inline pyperf/pyperf_run:20]:
```suggestion
	echo "--pyperf_version <version number>: Version of pyperf to run, default is $PYPERF_VERSION."
```

**Review by kdvalin — CHANGES_REQUESTED** [review]
Some things I missed the first round.

@frival Thoughts on this?

File: pyperf/pyperf_run [hunk 6]
```diff
@@ -245,15 +252,18 @@ done
...
 	rm -rf pyperformance
-	git clone https://github.com/python/pyperformance
-	if [ $? -ne 0 ]; then
-		exit_out "Cloning of https://github.com/python/pyperformance failed." 1
+	#
+	# Install pip/pip3
+	#
+	python3 -m ensurepip --upgrade
+	if [[ $? -ne 0 ]]; then
+		exit_out "Failed: python3 -m ensurepip --upgrade" 1
```
Comment (kdvalin) [inline pyperf/pyperf_run:260]: After digging around, RHEL and Ubuntu both use a `python3-pip` package.

Considering [this](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/html/installing_and_using_dynamic_programming_languages/assembly_installing-and-using-python_installing-and-using-dynamic-programming-languages#proc_installing-additional-python-3-packages_assembly_installing-and-using-python) product doc, we should be specifying `python3-pip` as a dependency.

**Review by kdvalin — CHANGES_REQUESTED** [review]

File: pyperf/pyperf_run [hunk 7]
```diff
@@ -4,6 +4,7 @@ PATH="${PATH}:/usr/local/bin"
 export PATH
 python_pkgs=""
 python_exec=""
+PYPERF_VERSION="1.0.4"
```
Comment (kdvalin) [inline pyperf/pyperf_run:7]: Considering future python versions don't work well with this pyperf version, I think we can bump it up to the current latest `1.11.0`.

@frival feel free to object :)

**Review by kdvalin — CHANGES_REQUESTED** [review]

**Review by kdvalin — APPROVED** [review]
LGTM now.  @frival you want to double check I didn't clobber anything accidentally?

**Review by frival — APPROVED** [review]
Aside from the fact that the first commit ought to be split into two, LGTM

**Pull request merged**